cob_script_server/src/simple_script_server/simple_script_server.py

```diff
@@ -595,31 +599,68 @@ def compose_trajectory(self, component_name, parameter_name, speed_factor=1.0, u
 		point_nr = 0
 		traj_time = 0
 
-		param_string = self.ns_global_prefix + "/" + component_name + "/default_vel"
-		if not rospy.has_param(param_string):
-			default_vel = 0.1 # rad/s
-			rospy.logwarn("parameter %s does not exist on ROS Parameter Server, using default of %f [rad/sec].",param_string,default_vel)
-		else:
-			default_vel = rospy.get_param(param_string)
+		if default_vel: # passed via argument
```

Comment:
The branch complexity of this method is getting bigger and bigger. Maybe we should clean that up into at least a separate method for just determining the desired speed... In a next PR